packages/arb-token-bridge-ui/src/components/common/NetworkSelectionContainer.tsx

```diff
@@ -67,19 +84,49 @@ export const NetworkSelectionContainer = ({
           | undefined
       ) => void
     ) => {
-      switchNetwork?.(Number(chainId))
+      setSelectedChainId(chainId)
```

Comment:
I'm not sure if this is what's causing it, but when you're connected and switch network, the network container in the header flashes from: current network -> new network -> current network before showing the Metamask popup

Reply:
good catch, fixed it